aixplain/factories/team_agent_factory/__init__.py

```diff
@@ -68,8 +71,21 @@ def create(
                 assert isinstance(agent, Agent), "TeamAgent Onboarding Error: Agents must be instances of Agent class"
             agent_list.append(agent_obj)
 
-        if use_inspector and not use_mentalist:
-            raise Exception("TeamAgent Onboarding Error: To use the Inspector agent, you must enable Mentalist.")
+        # NOTE: backend expects max_inspectors (for "generated" inspectors)
+        max_inspectors = num_inspectors
+
+        if use_inspector:
```

Comment:
Please also check for any inconsistency in the received parameters, e.g. if `use_inspector` is False and `num_inspectors` > 1 or the `inspector_targets` is an empty list. Also, should `inspector_targets` have a length equal to `max_inpectors`?